frontend/src/App.tsx

```diff
@@ -1,41 +1,38 @@
-import { BrowserRouter, Route, Routes } from "react-router-dom";
-import SignUpPage from "./auth/pages/Signup";
-import SignInPage from "./auth/pages/Signin";
-import RootLayout from "./layouts/root-layout";
-import { AuthenticateWithRedirectCallback } from "@clerk/clerk-react";
-import Verify from "./auth/pages/Verify";
-import { ClerkProviderWrapper } from "./providers/ClerkProvider";
-import { ApolloProviderWrapper } from "./providers/ApolloProvider";
-import { AuthProvider } from "./auth/context/AuthContext";
-import JournalDetails from "./journal/pages/JournalsDetails";
-import RootJournal from "./journal/layouts/root-journal";
-import Continue from "./auth/pages/Continue";
+import { BrowserRouter, Route, Routes } from 'react-router-dom';
+import SignUpPage from './auth/pages/Signup';
+import SignInPage from './auth/pages/Signin';
+import RootLayout from './layouts/root-layout';
+import { AuthenticateWithRedirectCallback } from '@clerk/clerk-react';
+import Verify from './auth/pages/Verify';
+import { ClerkProviderWrapper } from './providers/ClerkProvider';
+import { ApolloProviderWrapper } from './providers/ApolloProvider';
+import { AuthProvider } from './auth/context/AuthContext';
+import JournalDetails from './journal/pages/JournalsDetails';
+import RootJournal from './journal/layouts/root-journal';
+import Continue from './auth/pages/Continue';
+import { FolderProvider } from './folder/context/FolderContext';
 
 export default function App() {
   return (
     <BrowserRouter>
       <ClerkProviderWrapper>
         <ApolloProviderWrapper>
           <AuthProvider>
-            <Routes>
-              <Route path="/" element={<RootLayout />}>
-                <Route path="journals" element={<RootJournal />}>
-                  <Route path=":id" element={<JournalDetails />} />
+            <FolderProvider>
```

Comment:
You only need folder provider as part of your journals routes

All the pages that are related to sign in don't need that context